src/internals/ir/callGraph.ts

```diff
@@ -248,10 +251,24 @@ export class CallGraph {
       if (isSelf(expr.self)) {
         if (currentContractName !== undefined) {
           return `${currentContractName}::${methodName}`;
+        } else if (astStore) {
+          const functions = Array.from(astStore.getFunctions());
+          for (const func of functions) {
+            if (
+              func.kind === "function_def" &&
+              idText(func.name) === methodName &&
+              this.isExtensionFunction(func)
+            ) {
+              const selfType = this.getExtensionSelfType(func);
+              if (selfType) {
+                return `${selfType}::${methodName}`;
+              }
+              break;
+            }
+          }
+          return methodName;
         } else {
-          throw InternalException.make(
-            `Cannot process ${pp(expr)} without current contract name`,
-          );
+          return methodName;
```

Comment:
This makes no sense. If we are here, we found a bug in Misti a should `throw` an error.